mgca: Add test for normalizing const projections with assoc const equality bounds

**EDIT(@camelid): If you're interested in doing this, please add a `//@ check-pass` test to `tests/ui/const-generics/mgca/` that is along the lines of the example in https://github.com/rust-lang/rust/issues/120905#issuecomment-3706467915. Feel free to ping me (@camelid) or assign me on your PR.**

-----

For example, I expected the following code to succeed compilation:

```rs
#![feature(generic_const_exprs)]
#![feature(associated_const_equality)]

trait Trait {
    const C: usize;
}

fn f<T: Trait<C = 1>>()
where
    [(); T::C]:
{
    let _: [(); T::C] = [()];
}
```

Since (I'm pretty sure that) the ParamEnv contains the projection predicate `<T as Trait>::C == 1`, the const projection `<T as Trait>::C` inside the function body should be able to get normalized to `1`. That however doesn't seem to happen:

```
error[E0308]: mismatched types
  --> src/lib.rs:12:25
   |
12 |     let _: [(); T::C] = [()];
   |                         ^^^^ expected `T::C`, found `1`
   |
   = note: expected constant `T::C`
              found constant `1`
```

--- a/tests/ui/const-generics/associated-const-bindings/normalization-via-param-env.rs
+++ b/tests/ui/const-generics/associated-const-bindings/normalization-via-param-env.rs
@@ -1,0 +1,18 @@
+//@ check-pass
+#![feature(min_generic_const_args)]
+#![allow(incomplete_features)]
+
+// Regression test for normalizing const projections
+// with associated const equality bounds.
+
+trait Trait {
+    #[type_const]
+    const C: usize;
+}
+
+fn f<T: Trait<C = 1>>() {
+    // This must normalize <T as Trait>::C to 1
+    let _: [(); T::C] = [()];
+}
+
+fn main() {}

